### PR TITLE
Add nemo v1.22.0

### PR DESCRIPTION
**Pull request by srishtih**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: nemo.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      # allow site-packages
+      sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.destdir}}/usr/share/nemo/.venv/pyvenv.cfg
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: NVIDIA/NeMo
+    strip-prefix: v
```
Comment (mamccorm) [inline nemo.yaml:75]: I think you need to also filter by 'v' here as there are other version tags such as 'stable' which may get built, like this one: https://github.com/NVIDIA/NeMo/releases/tag/stable

```
tag-filter: v
```

File: nemo.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  - runs: |
+      # edit the venv paths
+      rm -rf ${{targets.destdir}}/usr/share/nemo/.venv/bin/__pycache*
+      sed -i "s|/home/build|/usr/share/nemo|g" ${{targets.destdir}}/usr/share/nemo/.venv/bin/*
+
+  - runs: |
+      # allow site-packages
+      sed -i "s|include-system-site-packages = false|include-system-site-packages = true|g" ${{targets.destdir}}/usr/share/nemo/.venv/pyvenv.cfg
+
```
Comment (mamccorm) [inline nemo.yaml:70]: Do we need an extra step here similar to this package?
- https://github.com/wolfi-dev/os/blob/9e21b9522815e253e6fe89102a999186d1e06c49/az.yaml#L51

i.e something like:

```
  - runs: |
      mkdir -p ${{targets.destdir}}/usr/bin
      ln -s /usr/share/nemo/.venv/bin/nemo ${{targets.destdir}}/usr/bin/nemo
```

However if i build nemo and check in the `/usr/share/nemo/.venv/bin` directory, i don't see a 'nemo' executable. if I do the same with the AZ package I do.

File: nemo.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      repository: https://github.com/NVIDIA/NeMo.git
+      tag: v${{package.version}}
+      expected-commit: 0b7467ebd158ef6876dd147189517b25ef626982
+
+  - runs: git submodule update --init --recursive
+
+  - runs: |
+      sed -i '/fasttext/d' requirements/requirements_nlp.txt
+      sed -i 's/megatron_core==0.4.0/megatron_core==0.3.0/' requirements/requirements_nlp.txt
+
```
Comment (mamccorm) [inline nemo.yaml:45]: I know this is draft btw, so feel free to ignore this comment if not relevant, but looks like we're missing the build step?
 
i.e we're just setting up the virtualEnv and installing the dependencies, but not building nemo? Maybe this what your testing.

I guess that explains why there is no nemo package inside the virtualEnv (my comment below).

[See here](https://github.com/wolfi-dev/os/blob/9e21b9522815e253e6fe89102a999186d1e06c49/az.yaml#L26) for another example.

File: nemo.yaml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+      # Install transformer engine
+          # same as Apex
+
+      # pip3 uninstall -y sacrebleu torchtext most likely not needed as these came from nvidia-pytorch image
+
+      # export LHOTSE_REQUIRE_TORCHAUDIO=0
+
+      # install requirements from all requirements_xyz.txt
+      for f in ./requirements/requirements*.txt;
+        do pip3 install --disable-pip-version-check --no-cache-dir -r $f
```
Comment (smoser) [inline nemo.yaml:86]: this is actually done by setup.py below when '.[all]' is used.  the setup builds its requirements from "all" of the requirements/*.txt

**Pull request closed without merging**